File: solutions/java/knapsack/1/src/main/java/Knapsack.java
import java.util.List;
class Knapsack {
    int maximumValue(int maximumWeight, List<Item> items) {
        int[] maximumValue = {0};
        backtrack(maximumValue, maximumWeight, items, 0, 0, 0);
        return maximumValue[0];
    }
    private void backtrack(int[] maximumValue, int maximumWeight, List<Item> items, int startIndex, int currentWeight, int currentValue) {
        if (currentWeight <= maximumWeight) {
            maximumValue[0] = Math.max(maximumValue[0], currentValue);
        } else {
            return;
        }
        for (int i = startIndex; i < items.size(); i++) {
            backtrack(maximumValue, maximumWeight, items, i + 1, currentWeight + items.get(i).weight, currentValue + items.get(i).value);
        }
    }
}

File: solutions/julia/knapsack/1/knapsack.jl
function maximum_value(maxw, items)
    memo = zeros(Int, (length(items)+1, maxw+1))
    for j in 0:maxw, i in 1:length(items)
        memo[i+1, j+1] = items[i].weight > j ? memo[i, j+1] : max(memo[i, j+1], memo[i, j+1-items[i].weight] + items[i].value)
    end
    maximum(memo)
end